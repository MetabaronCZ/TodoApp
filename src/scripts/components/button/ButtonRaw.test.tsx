import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import { withMockedProviders } from 'test/component';
import { ButtonRaw } from 'components/button/ButtonRaw';

describe('components/button/ButtonRaw', () => {
  it('should render correctly', () => {
    const tree = render(
      withMockedProviders(
        <ButtonRaw>
          <span>CONTENT</span>
        </ButtonRaw>,
      ),
    );
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByText('CONTENT')).toBeInTheDocument();
  });

  it('should render props', () => {
    const { container } = render(
      withMockedProviders(
        <ButtonRaw id="test-id" className="test-class" title="Test button" />,
      ),
    );
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('test-class');
    expect(container.firstChild).toHaveAttribute('id', 'test-id');
    expect(container.firstChild).toHaveAttribute('title', 'Test button');
  });

  it('should render disabled state', () => {
    const { container } = render(withMockedProviders(<ButtonRaw disabled />));
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toBeDisabled();
  });

  it('should call onClick event', () => {
    const onClick = vi.fn();

    const tree = render(withMockedProviders(<ButtonRaw onClick={onClick} />));
    expect(onClick).not.toBeCalled();

    fireEvent.click(tree.getByRole('button'));
    expect(onClick).toBeCalledTimes(1);
    expect(onClick).lastCalledWith();
  });
});
