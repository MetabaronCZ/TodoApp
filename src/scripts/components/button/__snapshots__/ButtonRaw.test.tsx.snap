// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/button/ButtonRaw should render children 1`] = `
.c0 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

<div>
  <button
    class="c0"
    type="button"
  >
    <span>
      Test button
    </span>
  </button>
</div>
`;

exports[`components/button/ButtonRaw should render correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

<div>
  <button
    class="c0"
    type="button"
  />
</div>
`;

exports[`components/button/ButtonRaw should render disabled state 1`] = `
.c0 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

<div>
  <button
    class="c0"
    disabled=""
    type="button"
  />
</div>
`;

exports[`components/button/ButtonRaw should render props 1`] = `
.c0 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

<div>
  <button
    class="c0 test-class"
    id="test-id"
    title="Test button"
    type="button"
  />
</div>
`;
